### PR TITLE
fix: Fix NavigationView memory leak

### LINKED ISSUE #10309: Control leaks roundup
### Current behavior

This issue collects controls which are currently leaking on various platforms in the `When_Add_Remove` runtime test:

- [x] `Microsoft.UI.Xaml.Controls.AnimatedIcon` (WASM, Skia)
- [x] `Microsoft.UI.Xaml.Controls.BreadcrumbBar` (WASM, Skia)
- [x] `Microsoft.UI.Xaml.Controls.Expander` (WASM, Skia)
- [ ] `Microsoft.UI.Xaml.Controls.NavigationView` (WASM, Skia)

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

WebAssembly, Android, iOS, macOS (AppKit), Mac Catalyst, Skia (WPF), Skia (GTK on Linux/macOS/Windows), Skia (Linux Framebuffer), Skia (Tizen)

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #10309

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsRepeater.cs [hunk 1]
```diff
@@ -845,12 +845,12 @@ void OnLayoutChanged(Layout oldValue, Layout newValue)
 			if (newValue != null)
 			{
 				newValue.InitializeForContext(GetLayoutContext());
-				newValue.MeasureInvalidated += InvalidateMeasureForLayout;
-				newValue.ArrangeInvalidated += InvalidateArrangeForLayout;
+				newValue.WeakMeasureInvalidated += InvalidateMeasureForLayout;
```
Comment (MartinZikmund) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsRepeater.cs:848]: Please add a comment `// Uno specific:` to make it clear that the code intentionally differs here

Comment (Youssef1313) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsRepeater.cs:849]: Done.

**Review by jeromelaban — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsRepeater.cs [hunk 2]
```diff
@@ -845,12 +869,27 @@ void OnLayoutChanged(Layout oldValue, Layout newValue)
...
+				// Uno specific: Using weak subscription..
+				// ItemsRepeater uses a singleton StackLayout.
+				// Subscribing to MeasureInvalidated and failing to unsubscribe is a large memory leak.
+				// Unsubscribing for that in Unloaded failed, because subscription can happen before Loaded and in cases where Loaded/Unloaded are never raised.
+				// The fact that we subscribe in such case feels like a lifecycle bug related to applying templates (things are
+				// initiated in NavigationView.OnApplyTemplate, but probably that OnApplyTemplate call shouldn't have happened).
+				// For now, the only feasible solution is to have a weak event subscription.
+				// NOTE that at the time of writing this, there is another bad subscription that happens early in ItemsRepeater constructor, which isn't the case on WinUI.
+				// However, fixing that bad subscription will still leak due to the lifecycle issue (at least, at the time of writing this).
+
```
Comment (Youssef1313) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsRepeater.cs:881]: This is quite wrong. The root cause isn't a lifecycle bug.

It's the fact that the leaking ItemsRepeater is in a Flyout in NavigationView template. So, it won't get loaded without opening the flyout.

The proper fix is replicating what Fluent v2 styles do, i.e, not use the singleton `StackLayout` for this specific `ItemsRepeater`.

**Comment by Youssef1313** [comment]
Replaced by https://github.com/unoplatform/uno/pull/15037

**Pull request closed without merging**